src/kudu/consensus/raft_consensus.cc

```diff
@@ -877,9 +880,14 @@ void RaftConsensus::ReportFailureDetectedTask() {
   std::unique_lock<simple_spinlock> try_lock(failure_detector_election_lock_,
                                              std::try_to_lock);
   if (try_lock.owns_lock()) {
-    WARN_NOT_OK(StartElection(FLAGS_raft_enable_pre_election ?
-        PRE_ELECTION : NORMAL_ELECTION, { ELECTION_TIMEOUT_EXPIRED }),
-                LogPrefixThreadSafe() + "failed to trigger leader election");
+    MonoTime failureTime =
```

Comment:
can you add a comment here? The failure_detector_last_snoozed_ was the last time the current node snoozed.
Are we saying that we expected to hear atleast a heartbeat later... so that is the time of failure?